beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -293,17 +313,195 @@ func (f *blocksFetcher) fetchBlocksFromPeer(
 		Step:      1,
 	}
 	for i := 0; i < len(peers); i++ {
-		blocks, err := f.requestBlocks(ctx, req, peers[i])
-		if err == nil {
-			f.p2p.Peers().Scorers().BlockProviderScorer().Touch(peers[i])
-			return blocks, peers[i], nil
-		} else {
-			log.WithError(err).Debug("Could not request blocks by range")
+		p := peers[i]
+		blocks, err := f.requestBlocks(ctx, req, p)
+		if err != nil {
+			log.WithField("peer", p).WithError(err).Debug("Could not request blocks by range from peer")
+			continue
+		}
+		f.p2p.Peers().Scorers().BlockProviderScorer().Touch(p)
+		robs, err := validROBlocks(blocks)
+		if err != nil {
+			log.WithField("peer", p).WithError(err).Debug("invalid BeaconBlocksByRange response")
+			continue
 		}
+		return robs, p, err
 	}
 	return nil, "", errNoPeersAvailable
 }
 
+func validROBlocks(blocks []interfaces.ReadOnlySignedBeaconBlock) ([]BlockWithVerifiedBlobs, error) {
+	robs := make([]blocks2.ROBlock, len(blocks))
+	for i, b := range blocks {
+		rb, err := blocks2.NewROBlock(b)
+		if err != nil {
+			return nil, err
+		}
+		robs[i] = rb
+	}
+	sort.Sort(blocks2.ROBlockSlice(robs))
+	rb := make([]BlockWithVerifiedBlobs, len(robs))
+	for i := range robs {
+		rb[i] = BlockWithVerifiedBlobs{block: robs[i]}
```

Comment:
It's a little confusing to me why we need to make the slice `robs`  to then sort it to then copy the contents into the slice `rb`. Couldn't we simply implement the sort interface in `[]BlockWithVerifiedBlobs` and then create directly the slice of verified blocks?